File: backend/src/main/java/com/eshop/service/OrderService.java
package com.eshop.service;

import com.eshop.entity.Order;
import java.util.List;

public interface OrderService {
    List<Order> getAllOrders();

    Order getOrderById(Long id);

    List<Order> getOrdersByUserId(Long userId);

    Order createOrder(Order order);

    Order updateOrderStatus(Long id, String status);
}

File: frontend/src/services/orderService.js
import api from './api';

export const orderService = {
    // Get all orders (admin)
    getAllOrders: async () => {
        const response = await api.get('/orders');
        return response.data;
    },

    // Get order by ID
    getOrderById: async (id) => {
        const response = await api.get(`/orders/${id}`);
        return response.data;
    },

    // Get orders by user ID
    getOrdersByUserId: async (userId) => {
        const response = await api.get(`/orders/user/${userId}`);
        return response.data;
    },

    // Create order
    createOrder: async (orderData) => {
        const response = await api.post('/orders', orderData);
        return response.data;
    },

    // Update order status (admin)
    updateOrderStatus: async (id, status) => {
        const response = await api.put(`/orders/${id}/status`, { status });
        return response.data;
    }
};
